src/init.c

```diff
@@ -733,15 +733,17 @@ _dispatch_logv_init(void *context DISPATCH_UNUSED)
 			dispatch_log_basetime = _dispatch_absolute_time();
 #endif
 #if defined(_WIN32)
-			FILE *pLogFile = _fdopen(dispatch_logfile, "w");
-
 			char szProgramName[MAX_PATH + 1] = {0};
 			GetModuleFileNameA(NULL, szProgramName, MAX_PATH);
 
-			fprintf(pLogFile, "=== log file opened for %s[%lu] at "
-					"%ld.%06u ===\n", szProgramName, GetCurrentProcessId(),
-					tv.tv_sec, (int)tv.tv_usec);
-			fclose(pLogFile);
+			char szMessage[512];
+			int len = snprintf(szMessage, sizeof(szMessage),
+					"=== log file opened for %s[%lu] at %ld.%06u ===\n",
+					szProgramName, GetCurrentProcessId(), tv.tv_sec,
+					(int)tv.tv_usec);
+			if (len > 0 && (size_t)len < sizeof(szMessage)) {
```

Comment:
If the second check fails, wouldn't it be better to write the first sizeof(szMessage) bytes to the log rather than discard everything?